CMakeLists.txt

```
@@ -42,65 +52,68 @@ if (SINGULARITY_SUBMODULE_MODE)
+# well as later info printing and debugging
+set(SINGULARITY_PUBLIC_DEFINES "")  # list to `target_compile_definitions()`
+set(SINGULARITY_PRIVATE_DEFINES "") # """
+set(SINGULARITY_PUBLIC_LIBS "")      # list to `target_link_libraries()`
+set(SINGULARITY_PRIVATE_LIBS "")     # """
+set(SINGULARITY_EXPORT_TARGETS "")
+set(SINGULARITY_COMPILEFLAGS "")    # TODO: actually use these
+set(SINGULARITY_DEP_PKGS "")        # a list of packages, used to generate import rules for installed targets. (may want to rename)
 
-# TODO: Is this generator expression correct/needed?
-target_include_directories(singularity-eos::flags
-                           INTERFACE
-                           $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-                           $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/utils>)
+#------------------------------------------------------------------------------#
+# Compiler & language setup
+#------------------------------------------------------------------------------#
 
-# xl fix
-target_compile_options(singularity-eos::flags INTERFACE
-                       $<$<COMPILE_LANG_AND_ID:CXX,XL>:-std=c++1y;-qxflag=disable__cplusplusOverride>)
-target_link_options(singularity-eos::flags INTERFACE
-                   $<$<COMPILE_LANG_AND_ID:CXX,XL>:-std=c++1y;-qxflag=disable__cplusplusOverride>)
+# checks if this is our build, or we've 
+# been imported via `add_subdirectory`
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
 
-#target_compile_features(singularity-eos::flags INTERFACE cxx_std_14)
-#target_compile_features(singularity-eos::libs INTERFACE cxx_std_14)
+enable_language(CXX)
+include(CMakeDetermineCXXCompiler)
+if(SINGULARITY_USE_FORTRAN)
+  enable_language(Fortran)
+  include(CMakeDetermineFortranCompiler)
+endif()
+
+# NOTE: this may not work with kokkos, we'll see.
+#if(SINGULARITY_USE_CUDA)
+#  enable_language(CUDA)
+#  include(CMakeDetermineCUDACompiler)
+#endif()
+
+#------------------------------------------------------------------------------#
+# De-thaw some options
+#------------------------------------------------------------------------------#
 
-# check for currently incompatible option
-if(SINGULARITY_USE_CUDA AND NOT SINGULARITY_USE_KOKKOS)
-  message(FATAL_ERROR "Cuda without kokkos is not currently supported")
+# using this space for "immediate" logic that can be avoid messages getting
+# lost inside the brrrrrrrr of cmake.
+if(SINGULARITY_USE_KOKKOSKERNELS AND NOT SINGULARITY_USE_KOKKOS)
+  message(FATAL_ERROR "Kokkos is required for KokkosKernels.")
+endif()
+
+if(SINGULARITY_USE_CUDA)
+  if(NOT SINGULARITY_USE_KOKKOS)
+    message(FATAL_ERROR "CUDA without Kokkos is not currently supported. If available, reconfigure with '-DSINGULARITY_USE_KOKKOS=ON'. Otherwise, disable CUDA offloading '-DSINGULARITY_USE_CUDA=OFF'")
+  endif()
+  if(NOT SINGULARITY_USE_KOKKOSKERNELS)
```

```suggestion
 if(NOT SINGULARITY_USE_KOKKOSKERNELS AND SINGULARITY_BUILD_CLOSURE)
```
Linear algebra is only ever needed with BUILD_CLOSURE enabled.

You could probably disable the find package for the linear algebra if build closure is not enabled.